packages/flutter/lib/src/material/material_state.dart

```diff
@@ -309,26 +301,32 @@ abstract class MaterialStateOutlineInputBorder extends OutlineInputBorder implem
   /// Creates a [MaterialStateOutlineInputBorder] from a [MaterialPropertyResolver<InputBorder>]
   /// callback function.
   ///
-  /// If used as a regular input border, the border resolved in the default state (the
-  /// empty set of states) will be used.
+  /// [MaterialStateOutlineInputBorder] should only be used with widgets
+  /// that document their support, like [InputDecoration.border]
+  /// (if used as a regular [InputBorder], it acts the same as
+  /// an empty `OutlineInputBorder()` constructor).
```

Comment:
```suggestion
  /// If used as a regular [InputBorder], it acts the same as
  /// an empty `OutlineInputBorder()` constructor.
```
nit: prefer to have this as a normal sentence.